File: languages/php/Bloock/Entity/Anchor.php
<?php

namespace Bloock\Entity;

use Google\Protobuf\Internal\RepeatedField;

class Anchor
{
    private int $id;
    private RepeatedField $blockRoots;
    private RepeatedField $networks;
    private string $root;
    private string $status;

    public function __construct(int $id, RepeatedField $blockRoots, RepeatedField $networks, string $root, string $status) {
        $this->id = $id;
        $this->blockRoots = $blockRoots;
        $this->networks = $networks;
        $this->root = $root;
        $this->status = $status;
    }

    public static function fromProto(\Bloock\Anchor $anchor): Anchor {
        return new Anchor($anchor->getId(), $anchor->getBlockRoots(), $anchor->getNetworks(), $anchor->getRoot(), $anchor->getStatus());
    }

    public function toProto(): \Bloock\Anchor {
        $p = new \Bloock\Anchor();
        $p->setId($this->id);
        $p->setBlockRoots($this->blockRoots);

        $n = [];
        foreach ($this->networks->getIterator() as $network) {
            $n[] = $network->toProto();
        }
        $p->setNetworks($n);
        return $p;
    }

    /**
     * @return int
     */
    public function getId(): int
    {
        return $this->id;
    }

    /**
     * @return RepeatedField
     */
    public function getBlockRoots(): RepeatedField
    {
        return $this->blockRoots;
    }

    /**
     * @return RepeatedField
     */
    public function getNetworks(): RepeatedField
    {
        return $this->networks;
    }

    /**
     * @return string
     */
    public function getRoot(): string
    {
        return $this->root;
    }

    /**
     * @return string
     */
    public function getStatus(): string
    {
        return $this->status;
    }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/integrity/Anchor.java
package com.bloock.sdk.entity.integrity;

import com.bloock.sdk.bridge.proto.BloockIntegrityEntities;
import java.util.List;
import java.util.stream.Collectors;

/**
 * Represents information about an anchor.
 */
public class Anchor {
  long id;
  List<String> blockRoots;
  List<AnchorNetwork> networks;
  String root;
  String status;

  /**
   * Constructs an Anchor object with the specified parameters.
   * 
   * @param id
   * @param blockRoots
   * @param networks
   * @param root
   * @param status
   */
  Anchor(
      long id, List<String> blockRoots, List<AnchorNetwork> networks, String root, String status) {
    this.id = id;
    this.blockRoots = blockRoots;
    this.networks = networks;
    this.root = root;
    this.status = status;
  }

  public static Anchor fromProto(BloockIntegrityEntities.Anchor anchor) {
    return new Anchor(
        anchor.getId(),
        anchor.getBlockRootsList(),
        anchor.getNetworksList().stream()
            .map(AnchorNetwork::fromProto)
            .collect(Collectors.toList()),
        anchor.getRoot(),
        anchor.getStatus());
  }

  BloockIntegrityEntities.Anchor toProto() {
    return BloockIntegrityEntities.Anchor.newBuilder()
        .setId(this.id)
        .addAllBlockRoots(this.blockRoots)
        .addAllNetworks(
            this.networks.stream().map(AnchorNetwork::toProto).collect(Collectors.toList()))
        .build();
  }

  @Override
  public String toString() {
    return "{"
        + "id: "
        + id
        + ", blockRoots: "
        + blockRoots
        + ", root: "
        + root
        + ", status: "
        + status
        + ", networks: "
        + networks
        + "}";
  }

  /**
   * Gets id from the anchor.
   * 
   * @return
   */
  public long getId() {
    return id;
  }

  /**
   * Gets block roots from the anchor.
   * 
   * @return
   */
  public List<String> getBlockRoots() {
    return blockRoots;
  }

  /**
   * Gets networks from the anchor.
   * 
   * @return
   */
  public List<AnchorNetwork> getNetworks() {
    return networks;
  }

  /**
   * Gets root from the anchor.
   * 
   * @return
   */
  public String getRoot() {
    return root;
  }

  /**
   * Get status from the anchor.
   * 
   * @return
   */
  public String getStatus() {
    return status;
  }
}
